backend/scripts/github_collector.py

```
@@ -284,26 +428,289 @@ def collect_all(self):
+                "error": str(e)
+            }
+    
+    def get_cached_branches_from_mongodb(self, repo_name: str) -> List[Dict]:
+        """Get list of branches from MongoDB metadata (Cache-First)"""
+        if self.mongo_db is None and self.mongo_client is None:
+             # Try connecting if not already connected
+             try:
+                mongodb_uri = os.getenv("MONGODB_URI", "mongodb://localhost:27017/contextkeeper")
+                self.mongo_client = MongoClient(mongodb_uri, serverSelectionTimeoutMS=2000)
+                self.mongo_db = self.mongo_client.get_database()
+             except:
```

Ruff (0.14.8)

Do not use bare `except`

(E722)